### DISCUSSION TITLE
macOS 11.7.10: ISO C++17 error under Big Sur

### DISCUSSION THREAD

**Discussion by steamfan** [brndnmtthws/conky#1865] [Q&A] [answered]
Hello,
I donwloaded conky sources from GitHub and followed the instructions under [MacOS Build](https://github.com/brndnmtthws/conky/wiki/Installation#macos-build).
At the last step` cmake --build . `I get these error messages:

[ 17%] Building CXX object src/CMakeFiles/conky.dir/colours.cc.o
In file included from /System/Volumes/Data/NAS/Applications/conky/conky/src/colours.cc:45:
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:63:20: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
gperf_case_strcmp (register const char *s1, register const char *s2)
                   ^~~~~~~~~
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:63:45: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
gperf_case_strcmp (register const char *s1, register const char *s2)
                                            ^~~~~~~~~
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:85:24: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
color_name_hash::hash (register const char *str, register unsigned int len)
                       ^~~~~~~~~
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:85:50: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
color_name_hash::hash (register const char *str, register unsigned int len)
                                                 ^~~~~~~~~
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:103:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register unsigned int hval = len;
  ^~~~~~~~~
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:149:31: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
color_name_hash::in_word_set (register const char *str, register unsigned int len)
                              ^~~~~~~~~
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:149:57: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
color_name_hash::in_word_set (register const char *str, register unsigned int len)
                                                        ^~~~~~~~~
/System/Volumes/Data/NAS/Applications/conky/conky/build/colour-names.cc:1556:11: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
          register const char *s = wordlist[key].name;

How can I solve those errors?

**Comment by brndnmtthws** [comment, marked as answer]
Try installing gperf from homebrew with `brew install gperf`, then recompile Conky, and you should be good to go.

It looks like you hit an issue with older versions of gperf: https://savannah.gnu.org/bugs/?53028.

Most likely it's because macOS include gperf, and your version is unpatched. It's fixed in gperf 3.1 (https://git.savannah.gnu.org/gitweb/?p=gperf.git;a=commit;h=baf89d87f0d94b83e7764569da9d84dd36f82ba9), which is available from Homebrew: https://formulae.brew.sh/formula/gperf.

We can perhaps disable the warning when building that particular file, however, to make it compile with older versions.

**Reply by brndnmtthws** [reply]
I updated the Wiki to include gperf in the dependency list.

**Reply by steamfan** [reply]
Thanks for  your support. This solved the issue.
